crates/http/src/routes.rs

```diff
@@ -291,13 +278,45 @@ impl RouteMatch {
     }
 
     /// The named wildcards captured from the path, if any
-    pub fn named_wildcards(&self) -> &HashMap<String, String> {
-        &self.named_wildcards
+    pub fn named_wildcards(&self) -> HashMap<String, String> {
+        let Some((_, best_match)) = &self.best_match else {
+            return HashMap::new();
+        };
+        best_match
+            .captures()
+            .iter()
+            .map(|(k, v)| (k.to_owned(), v.to_owned()))
+            .collect()
     }
 
     /// The trailing wildcard part of the path, if any
     pub fn trailing_wildcard(&self) -> String {
-        self.trailing_wildcard.clone().unwrap_or_default()
+        // If we have a trailing wildcard, return it.
+        if let Some(wildcard) = &self.trailing_wildcard {
+            return wildcard.clone();
+        };
+
+        // Otherwise, we need to extract it from the best match if we have it.
+        let Some((path, best_match)) = &self.best_match else {
+            return String::new();
+        };
+        best_match
+            .captures()
```

Comment:
Is it worth further optimising this by running `Match::captures()` only once, e.g. during the construction of the `RouteMatch` object?  The `captures()` implementation is non-trivial - I'm not sure what the lifetime considerations would be but it seems like you have the `'router` and `'path` lifetimes already.

I realise this is existing code, just raising it since we are in optimisation mode!

Reply:
Done! 🎉